webapp/packages/plugin-data-viewer/src/DatabaseDataModel/IDatabaseDataSource.ts

```diff
@@ -98,4 +99,5 @@ export interface IDatabaseDataSource<TOptions, TResult extends IDatabaseDataResu
   clearError: () => this;
   resetData: () => this;
   dispose: () => Promise<void>;
+  closeResults?: (results: IDatabaseResultSet[]) => Promise<void>;
```

Comment:
please remove it from base database data source, even more you can't use *ResultSet types in this base interface

ResultSetDataSource is better place for it